### DISCUSSION TITLE
useSupabaseClient() realtime channel not updating at first

### DISCUSSION THREAD

**Discussion by spoken-angus** [supabase/supabase#11399] [Questions] [answered]
I'm working on a Next.js project and implemented a real-time channel for a page. However, the first few seconds that the page is loaded, the channel is not receiving any postgres changes even when I make them. It does eventually work but I am confused why it isn't immediately realtime. 

```
const supabase = useSupabaseClient()

useEffect(() => {
    supabase
      .channel(`count:${query.slug}`)
      .on(
        "postgres_changes",
        { event: "INSERT", schema: "public", table: "counts" },
        (payload: any) => {
          setCount((count) => count + (payload.new.inout ? 1 : -1));
          });
        }
      )
      .subscribe();
  }, []);
```

**Comment by GaryAustin1** [comment, marked as answer]
There is a message sent when the connection is "really" connected.   
This issue covered it.  Towards the end is a method to check, BUT I've not verified it recently and it kept changing....
https://github.com/supabase/realtime/issues/282
